spec/operators/catchError-spec.ts

```diff
@@ -477,4 +451,57 @@ describe('catchError operator', () => {
     expect(sideEffects).to.deep.equal([0, 1, 2]);
   });
 
+  it('should call the cleanup function of the catched observable when the source throws synchronously', () => {
+    let clean = false;
+    const testCleanup$ = new Observable(() => () => {
+      clean = true;
+    })
+
+    const subscription = throwError(new Error('Some error')).pipe(
+      catchError(() => testCleanup$),
+    ).subscribe()
+    expect(clean).to.equal(false);
+
+    subscription.unsubscribe()
+    expect(clean).to.equal(true);
+  });
+
+  it('should not alter the order of the teardown logic of the subscription chain', () => {
```

Comment:
This test is a little too clever.

I'd prefer that things were more clearly named. For example, you're leveraging `i` in that for loop, but from the name it's not clear what the intended use of `i` is, or why it lives outside of the `for` declaration. You have to study the code to realize its part of the test, because you're throwing if you "make it through the loop". That test is probably unnecessary BTW.

Basically just an `of(1, 2, 3, 4, 5, 6, 7)` or something would probably be fine.

Reply:
Yep, I got a bit lazy and I just made a test out of [your comment](https://github.com/ReactiveX/rxjs/pull/5666#pullrequestreview-475053120) :see_no_evil: 

I will improve that, for sure. Thanks!

> That test is probably unnecessary BTW.

I don't know, I think that it doesn't hurt, with my previous implementation all the tests were passing, but this test would have failed... Let me try to make it more explicit and understandable.